include/rocksdb/options.h

```diff
@@ -1348,6 +1332,22 @@ struct DBOptions {
   std::shared_ptr<Cache> row_cache;
 
 #ifndef ROCKSDB_LITE
```

Comment:
We should avoid flag in public header, as users usually won't have it set.

Usually what we do is to leave the interface of SstFileManager, and implement a NewSstFileManager() that returns error. (Or null in this case).